### ISSUE TITLE
Bug: `layer(geom = "boxplot",...)` produces error

### ISSUE THREAD

**Issue by davidhodge931** [tidyverse/ggplot2#6191]
Wondering if it was possible to make the extra args (.e.g.  median.colour) added to boxplot could also available via `layer`?

``` r
library(tidyverse)
library(palmerpenguins)

penguins |>
  ggplot(aes(    y = flipper_length_mm,
                 x = species)) +
  geom_boxplot(
    median.colour = "red",
  )
#> Warning: Removed 2 rows containing non-finite outside the scale range
#> (`stat_boxplot()`).
```

![](https://i.imgur.com/R6QGO6c.png)

``` r

penguins |>
  ggplot(aes(    y = flipper_length_mm,
                 x = species)) +
  layer(
    geom = "boxplot",
    stat = "boxplot", 
    position = "dodge2",
    params = list(median.colour = "red"),
  )
#> Warning: Ignoring unknown parameters: `median.colour`
#> Removed 2 rows containing non-finite outside the scale range
#> (`stat_boxplot()`).
```

![](https://i.imgur.com/pTc6sOk.png)

<sup>Created on 2024-11-20 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by teunbrand** [comment]
The answer on whether this is possible is 'yes'. 
My answer on whether this should happen is 'probably not'.
There are just too many dang arguments to pass around.
As the one who implemented these, I'm a bit biased of course.
I'm happy to implement something else if you can get one of the other maintainers on board.

**Comment by davidhodge931** [comment]
Ahh, I see.. 

Maybe related to this is that layer seem to error for boxplot with a colour/fill aesthetic?

``` r
library(tidyverse)
library(palmerpenguins)

penguins |>
  ggplot(aes(x = species, y = flipper_length_mm, colour = species)) +
  layer(
    geom = "boxplot",
    stat = "boxplot",
    position = "dodge2",
  )
#> Warning: Removed 2 rows containing non-finite outside the scale range
#> (`stat_boxplot()`).
#> Error in unit(x, default.units): 'x' and 'units' must have length > 0
```

<sup>Created on 2024-11-20 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by teunbrand** [comment]
You're right about that related case and I'm happy to fix that to ensure `layer(geom = "boxplot")` works

**Comment by davidhodge931** [comment]
Do you want to leave this open as issue (or create an issue) that the extra new boxplot params can't be used with `layer`? 

Or just close for now once the bug bit has been fixed in that it is not that important and can't be done any time soon? 

(My bad for mixing 2 issues)

**Comment by teunbrand** [comment]
> the extra new boxplot params can't be used with layer

They can, but they should be lists of `colour`/`linetype`/`linewidth` parameters, as `geom_boxplot()` handles them internally.
If you'd like to keep this issue open to get a 2nd opinion, that is fine with me.
If not, I can close it whenever #6192 gets merged.

**Comment by davidhodge931** [comment]
Ah, I get it now - nice!
